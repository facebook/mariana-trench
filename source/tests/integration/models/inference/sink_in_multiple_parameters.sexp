(method (private) "LFlow;.flow:(LData;LData;)V"
 (
  (load-param v0)
  (load-param v1)
  (load-param v2)

  LINE(1)
  (invoke-virtual (v0 v1) "LSink;.sink:(LData;)V")

  LINE(3)
  (invoke-virtual (v0 v2) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
