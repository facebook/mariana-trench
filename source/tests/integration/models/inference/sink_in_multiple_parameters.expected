{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "method" : "LFlow;.flow:(LData;LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 1,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        },
        {
          "call_position" :
          {
            "line" : 3,
            "path" : "Flow.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(2)",
          "distance" : 1,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    }
  ]
}
